File: bff-gateway/src/routes/health.ts
import type { FastifyPluginAsync } from 'fastify';
import { getLive, getReady } from '../controllers/health.controller.js';

const healthPlugin: FastifyPluginAsync = async (app) => {
  app.get('/live', {
    schema: {
      tags: ['Health'],
      summary: 'Liveness probe',
      description:
        'Returns HTTP 200 when the BFF Gateway process is running. ' +
        'Used by orchestrators (Kubernetes, Cloud Run) to decide whether to restart the container.',
      response: {
        200: {
          description: 'Service is alive.',
          type: 'object',
          properties: {
            status: {
              type: 'string',
              description: 'Always "ok" when the service is live.',
              example: 'ok',
            },
          },
        },
      },
    },
  }, getLive);

  app.get('/ready', {
    schema: {
      tags: ['Health'],
      summary: 'Readiness probe',
      description:
        'Returns HTTP 200 when the BFF Gateway is ready to accept traffic. ' +
        'Used by orchestrators to gate traffic until the service has fully initialised.',
      response: {
        200: {
          description: 'Service is ready to receive traffic.',
          type: 'object',
          properties: {
            status: {
              type: 'string',
              description: 'Always "ok" when the service is ready.',
              example: 'ok',
            },
          },
        },
      },
    },
  }, getReady);
};

export default healthPlugin;

File: apps/document-reader/app/routers/health.py
from bowltie_py import format_success
from fastapi import APIRouter
from fastapi.responses import JSONResponse

router = APIRouter(tags=["Health"])


@router.get("/live")
async def liveness_check() -> JSONResponse:
    """Liveness probe — confirms the service process is running.

    Used by orchestrators (Cloud Run, Kubernetes) to decide whether to restart
    the container.  Returns HTTP 200 as long as the process is alive.

    Returns:
        JSONResponse containing a Bowltie-formatted ``{"status": "ok"}`` payload.
    """
    return JSONResponse(format_success({"status": "ok"}))


@router.get("/ready")
async def readiness_check() -> JSONResponse:
    """Readiness probe — confirms the service is ready to accept traffic.

    Used by orchestrators to gate traffic until the service has fully
    initialised.  Returns HTTP 200 when all startup steps have completed.

    Returns:
        JSONResponse containing a Bowltie-formatted ``{"status": "ok"}`` payload.
    """
    return JSONResponse(format_success({"status": "ok"}))
